pkg/operator/ceph/object/objectstore.go

```diff
@@ -1000,3 +999,38 @@ func errorOrIsNotFound(err error, msg string, args ...string) error {
 	}
 	return errors.Wrapf(err, msg, args)
 }
+
+// check whether zone endpoint list need to be updated or not
+func UpdateZoneEndpointList(zones []zoneType, desiredEndpointList []string, zoneName string) bool {
+	updateZoneEndpointList := false
+	isZonePresent := false
+	zoneEndpointsList := []string{}
+	checkZoneEndpoints := make(map[string]bool)
+	if zoneName != "" {
```

Comment:
There is no need to encase the below code inside this if block. We should use fail-fast techniques to return early under non-allowable conditions like this one

```go
if zoneName == "" {
   return false, errors.Errorf(/* ... */)
}
```